pkg/util/yaml.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
```

Comment:
The general recommendation is to avoid `util` packages (see https://blog.golang.org/package-names). In the operators case the `util` package includes a lot of unrelated code. Nested packages under a common `util` directory may be a reasonable compromise, e.g. istio.io/operator/pkg/util/yaml.

Reply:
Created https://github.com/istio/istio/issues/15606 to track this.